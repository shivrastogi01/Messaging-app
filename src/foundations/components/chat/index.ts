export * from "./text-message/text-message";
export * from "./image-message/image-message";
export * from "./link-message/link-message";
export * from "./giphy-message/giphy-message";
export * from "./avatar/avatar";
