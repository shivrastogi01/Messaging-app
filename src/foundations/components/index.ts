export * from "./presentation";
export * from "./layout";
