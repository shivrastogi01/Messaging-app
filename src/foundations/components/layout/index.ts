export * from "./styled-box/styled-box";
export * from "./scroll-view/scroll-view";
export * from "./flex-row/flex-row";
export * from "./flex-column/flex-column";
export * from "./modal/modal";
export * from "./drawer/drawer";
export * from "./list-item/list-item";
export * from "./dropdown/dropdown";
