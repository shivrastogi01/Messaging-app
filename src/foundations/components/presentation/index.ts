export * from "./heading/heading";
export * from "./label/label";
export * from "./title/title";
export * from "./icon/icon";
export * from "./button/button";
export * from "./input/input";
export * from "./textarea/textarea";
