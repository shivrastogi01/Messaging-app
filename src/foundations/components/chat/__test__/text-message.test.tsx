import React from "react";
import { TextMessage } from "foundations/components/chat";
import renderWithTheme from "main/testing/render-with-theme";

describe("Text Message", () => {
  test("emoji only message matches the snapshot", () => {
    const { container } = renderWithTheme(<TextMessage text="ðŸ˜ƒðŸ˜ƒðŸ˜ƒ" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("regular text message matches the snapshot", () => {
    const { container } = renderWithTheme(
      <TextMessage text="This is a plain old text message." />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("doesn't mutate the text", () => {
    const text = `This is a random number ${Math.random() * 10}`;
    const { getByText } = renderWithTheme(<TextMessage text={text} />);
    expect(getByText(text)).toBeTruthy();
  });
});
