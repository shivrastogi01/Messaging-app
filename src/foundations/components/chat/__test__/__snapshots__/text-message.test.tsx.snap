// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Message emoji only message matches the snapshot 1`] = `
.c0 {
  background: #FFFFFF;
  border-radius: 10px;
  border-top-left-radius: 0;
  box-shadow: 0 6px 10px rgba(103,19,176,0.06);
  color: #585858;
  line-height: 1.5;
  padding: 16px;
  text-align: left;
  white-space: pre-wrap;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  word-break: break-word;
  font-size: 20px;
}

<div
  class="c0"
>
  ðŸ˜ƒðŸ˜ƒðŸ˜ƒ
</div>
`;

exports[`Text Message regular text message matches the snapshot 1`] = `
.c0 {
  background: #FFFFFF;
  border-radius: 10px;
  border-top-left-radius: 0;
  box-shadow: 0 6px 10px rgba(103,19,176,0.06);
  color: #585858;
  line-height: 1.5;
  padding: 16px;
  text-align: left;
  white-space: pre-wrap;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  word-break: break-word;
}

<div
  class="c0"
>
  This is a plain old text message.
</div>
`;
